### PR TITLE
fix: PublicShareSidebar for Nextcloud 31

### LINKED ISSUE #13402: Talk sidebar not shown for public shares in Nextcloud 31
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

In Nextcloud 31 the public share page for single files and for folders was unified [when it was migrated to Vue](https://github.com/nextcloud/server/pull/45652); now single file shares show a file list with a single file, but no Talk sidebar.

Note that in previous Nextcloud versions the Talk sidebar was shown for public shares of a single file, but not in folder shares.

Anyway, maybe there will be further changes to the public share pages (I think that @susnux mentioned something about that, like providing a file list sidebar, but I might be mixing things up, many chats during the conference :-) ), so it would be worth taking those possible changes into account when fixing this.

## Steps to reproduce
- In Talk administration settings, ensure that `Allow conversations on public shares for files` is enabled (it is by default, but just in case)
- Open the Files app
- Create a public share of a file
- Open the link

### Expected behaviour
There is a Talk sidebar

### Actual behaviour
There is no Talk sidebar, nor a way to show it

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* Fix #13402
  * Minor refactoring and restyling included

### 🖼️ Screenshots / Screencasts

🏚️ Before | 🏡 After
-- | --
image | image

### 🏁 Checklist

- [ ] 🌏 Tested with different browsers / clients:
  - [x] Chromium (Chrome / Edge / Opera / Brave)
  - [ ] Firefox
  - [ ] Safari
  - [ ] Talk Desktop
  - [ ] Not risky to browser differences / client
- [ ] 🖌️ Design was reviewed, approved or inspired by the design team
- [ ] ⛑️ Tests are included or not possible
- [ ] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

### CODE REVIEW

**Comment by nickvergessen** [comment]
Failing sharing tests are unrelated and fixed by https://github.com/nextcloud/spreed/pull/13632

File: src/PublicShareSidebarTrigger.vue [hunk 1]
```diff
@@ -4,62 +4,42 @@
...
 
-	props: {
-		sidebarState: {
-			type: Object,
-			required: true,
-		},
-	},
+const props = defineProps<{
+	sidebarState: UnwrapNestedRefs<{ isOpen: boolean }>
+}>()
```
Comment (DorraJaouad) [inline src/PublicShareSidebarTrigger.vue:28]: Why are you explicitly unrawpping this prop?

File: src/PublicShareSidebar.vue [hunk 2]
```diff
@@ -256,16 +256,18 @@ export default {
 body .modal-wrapper * {
 	box-sizing: border-box;
 }
+
+#content-vue:has(#talk-sidebar) ~ footer {
+	width: calc(100% - 2 * var(--body-container-margin) - clamp(300px, 27vw, 500px));
+}
```
Comment (DorraJaouad) [inline src/PublicShareSidebar.vue:262]: also mimic transition as it is being used in opening and collapsing sidebar.

File: src/mainPublicShareSidebar.js [hunk 3]
```diff
@@ -100,21 +80,27 @@ function addTalkSidebarTrigger() {
...
 /**
- *
+ * Mount the Talk sidebar next to the main content.
  */
-function getShareToken() {
-	const shareTokenElement = document.getElementById('sharingToken')
-	return shareTokenElement.value
+function addTalkSidebar() {
+	// location.pathname is expected to be like /index.php/s/<shareToken>
+	const shareToken = window.location.pathname.split('/').pop()
```
Comment (DorraJaouad) [inline src/mainPublicShareSidebar.js:87]: You can find it in initial state
image

File: src/PublicShareSidebarTrigger.vue (hunk 1 above)
Comment (Antreesy) [reply src/PublicShareSidebarTrigger.vue:28]: That's just the same type as declared for reactive()
image

**Comment by susnux** [comment]
> ⚠️ Not sure about the new way of getting shareToken, but previous method is no longer available

In the frontend just use `getSharingToken` from `@nextcloud/sharing/public`, this function works with all Nextcloud versions.
Ref: https://nextcloud-libraries.github.io/nextcloud-sharing/functions/public.getSharingToken.html

**Review by DorraJaouad — APPROVED** [review]
Tested 🦅

**Pull request merged**